Adds mini medical pouches

PurpleMartinJCK opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->
Not mapped in yet. Need second PR after this one with map stuff.

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/storage.dm:
```diff
@@ -322,7 +323,7 @@
 	update_ui_after_item_removal()
 
-/obj/item/weapon/storage/Initialize()
+/obj/item/weapon/storage/Initialize(var/special)
```
`/obj/item/weapon/storage/Initialize(var/map_loading, var/special)`

PsiOmegaDelta replied:
Also, when is this ever set or actually utilized?

PurpleMartinJCK replied:
It's set here https://github.com/Baystation12/Baystation12/pull/20805/files#diff-696e5b9c7016378c5e732c69564c93e3R80

comma replied:
That
is a spectaculary overcomplicated way to have specific items colored differently.
Can't you just loop over them in pouches init or something instead of touching base storage code and utilizing such a confusing web of calls that two devs couldn't see through it?

PurpleMartinJCK replied:
Yeah I'm doing something different with them in pouch's initialise

comma commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+	icon_state = icon_colour
+
+/obj/item/weapon/reagent_containers/hypospray/autoinjector/pouch_auto
+	name = "Emergency Autoinjector"
```
lowercase plox

comma commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+	desc = "Use and Wait for Medical"
+	var/icon_colour
+
+/obj/item/weapon/reagent_containers/hypospray/autoinjector/pouch_auto/New(var/location, var/given_colour)
```
are these color params ever used? These are created by storage's internal stuff since they're in starts_with

PurpleMartinJCK replied:
Yes, it's used https://github.com/Baystation12/Baystation12/pull/20805/files#diff-223d783c6a194ada3c25a865ad4dce8cR357

comma commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+/obj/item/weapon/reagent_containers/pill/pouch_pill/paracetamol
+	chem_type = /datum/reagent/paracetamol
+
+/obj/item/weapon/reagent_containers/pill/pouch_pill/New(var/location, var/icon_colour)
```
ditto

PurpleMartinJCK replied:
ditto https://github.com/Baystation12/Baystation12/pull/20805/files#diff-223d783c6a194ada3c25a865ad4dce8cR357

comma commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+		")
+
+/obj/item/weapon/reagent_containers/pill/pouch_pill
+	name = "Emergency Pill"
```
lowercase pls, gonna be used in a phrase
Also are subtypes not being named specifically intentional?

PurpleMartinJCK replied:
Yes, they're specifically not being labelled to disincentivise people nicking things out.

comma commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+	. = ..(base_icon)
+	name += ": [injury_type]"
+
+/obj/item/weapon/storage/med_pouch/verb/check_instruct()
```
I guarantee that in 2 months no one would remember this verb exists.
I think it would be better to have it be in examine, under a linkie like EC plaque does it.

PurpleMartinJCK replied:
Got a link or a path to search?

PurpleMartinJCK replied:
Found it, will use

PsiOmegaDelta commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+	if(opened)
+		icon_state = base_icon + "_t"
+	else
+		icon_state = base_icon
```
Or just 
```
	if(opened)
		icon_state = initial(icon_state) + "_t"
	else
		icon_state = initial(icon_state)
```

PurpleMartinJCK replied:
Why is using initial better than having a separate variable for it? I figured having the extra bit of safety with it was a good thing.

comma replied:
safety from what?
base_icons is what people used before discovery of initial more or less

PsiOmegaDelta commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+		)
+
+/obj/item/weapon/storage/med_pouch/Initialize()
+	. = ..(base_icon)
```
Just as a side-note, you're overriding the `map_loading` argument here.

PurpleMartinJCK replied:
Yep, I'll fix that with the above fix.

comma commented on code/game/objects/items/med_pouch.dm:
```diff
@@ -0,0 +1,152 @@
+
+/obj/item/weapon/storage/med_pouch
+	name = "emergency medical pouch"
+	desc = "For use in emergency situations only. <div style='font-weight: bold;'>Your location and identity will be broadcast</div>."
```
<div style='font-weight: bold;'>Your location and identity will be broadcast</div>.

Isn't a thing anymore hopefuly? Forgot to axe here?

PurpleMartinJCK replied:
Yep, I'll fix with the instruction change. Thanks.

PurpleMartinJCK commented:
Should be good again now!

mustafakalash commented on code/game/objects/items/weapons/storage/firstaid.dm:
```diff
@@ -109,6 +109,20 @@
 
+/obj/item/weapon/storage/firstaid/stab
+	name = "stabilise first aid"
+	desc = "Contains medical pouches and a stasis bag."
```
Description is not necessarily true. Also, the name is kind of weird. What about "stabilization first aid?"

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+		"}
+
+/obj/item/weapon/storage/med_pouch/examine()
+	..()
```
`. = ..()`

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+
+/obj/item/weapon/storage/med_pouch/examine()
+	..()
+	to_chat(usr, "For use in emergency situations only. <A href='?src=\ref[src];show_info=1'>Please read instructions before use.</A>")
```
Double description. Just send the link alone.

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+
+/obj/item/weapon/storage/med_pouch/update_icon()
+	if(opened)
+		icon_state = base_icon + "_t"
```
Might want to use the `base_icon_state` extension. Look at cameras or lighters for examples.

PurpleMartinJCK replied:
That just seems to be this but with extra steps and datums

PsiOmegaDelta replied:
`base_icon_state` is mainly for items that people want to be customized, and `initial()` is no longer applicable.
Which leads to the question, why isn't `initial(icon_state)` used here?

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+
+/obj/item/weapon/storage/med_pouch/attack_self(mob/user as mob)
+	if(!opened)
+		user.visible_message("<span class='notice'>\The [user] tears open \the [src.name], breaking the vacuum seal!</span>", "<span class='notice'>You tear open \the [src.name], breaking the vacuum seal!</span>")
```
`\the [src.name]` is identical to `[src]`

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+		storage_slots = 4
+		update_icon()
+	else
+		to_chat(user, "<span class='notice'>\The [src.name] is already ripped open.</span>")
```
Ditto on `\The [src.name]`

mustafakalash replied:
Does storage not open the UI on `attack_self`? If it does, you'll want to `..()` here instead, anyway.

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+
+/obj/item/weapon/reagent_containers/pill/pouch_pill
+	name = "emergency pill"
+	desc = "Take and wait for Medical"
```
Description wording is a bit weird.

mustafakalash commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,161 @@
+/obj/item/weapon/reagent_containers/pill/pouch_pill/paracetamol
+	chem_type = /datum/reagent/paracetamol
+
+/obj/item/weapon/reagent_containers/pill/pouch_pill/New()
```
`Initialize()`, unless there's a reason this must be `New()`

PurpleMartinJCK replied:
It has to be new for the reagents to be properly added.

PsiOmegaDelta replied:
Can confirm.
Something causes an empty `reagents` setup to be deleted somewhere in `/New()` but I've yet to be able to find where.

PurpleMartinJCK commented:
Updated

Techhead0 commented:
I kinda have a thing against "unlabeled medicine". You have multiple identical pills and multiple identical autoinjectors with different medicines wrapped up in the same pouch. Might as well have a single injector and a single pill each.

PurpleMartinJCK commented:
That would leave autoinjectors that are potentially more than 15 units in them. It seems a bit fast.

Techhead0 commented:
Thoughts on PR:
- All the pouches have the same name, instead of names like "emergency burn pouch" etc.
- Instructions being a var means you can actually have relevant instructions on each pouch instead of on-size fits all.
- Medicine should always be labeled somewhere what it is. A doctor unfamiliar with the specifics of the kit should not have to rip open and analyse a fresh one to find out what their patient took. 
- Comment above about identical-looking-but-different medicines. 
- A charcoal (carbon) pill would actually be useful for the toxins kit.

EDIT: IRC comments for the permanent record:
The entire pouch system treats players like babies who can't be trusted with basic medical supplies. You want people to not raid emergency supplies for regular use? Give them regular supplies for regular use.

PurpleMartinJCK commented:
Instructions are a var here, and the name is added to the kit.
Do you have another suggestion to prevent people wanting to go through a bunch and pick out the things they want?
The carbon pill will remove the dylovene pill from the stomach. The dylovene will prevent further injury while the person waits for medical help.

Techhead0 commented:
I feel like you didn't actually read my comment about instructions.

PurpleMartinJCK commented:
Ah, I thought you meant "Instructions, if they were to be a var, means you can actually have relevant instructions on each pouch instead of one-size fits all." but you meant "Instructions, being a var, means you could now have relevant instructions on each pouch instead of one-size fits all."

Techhead0 reviewed (dismissed):
On further read, Deletrathol is actually a deal-breaker. An unlabeled autoinjector given to people without medical training, with a notable adverse interaction with alcohol. There are no warnings to this possibility. And it is provided as one of three unlabeled orange auto-injectors in the same pouch.

PurpleMartinJCK committed:
Adds medical pouches
- intented as single use things
- open before use

PurpleMartinJCK commented:
@Techhead0 
Updated instructions with unique behaviour
Squashed as requested
Changed deletrathol from a child of tramadol to a stand-alone reagent
Added labelling to pills and autoinjectors
Added better names to the pouches

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,207 @@
+/obj/item/weapon/storage/med_pouch/CanUseTopic()
+	return STATUS_INTERACTIVE
+
+/obj/item/weapon/storage/med_pouch/OnTopic(href, href_list)
```
`/obj/item/weapon/storage/med_pouch/OnTopic(var/user, var/list/href_list)`

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,207 @@
+
+/obj/item/weapon/storage/med_pouch/OnTopic(href, href_list)
+	if(href_list["show_info"])
+		to_chat(usr, instructions)
```
`to_chat(user, instructions)`

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/med_pouch.dm:
```diff
@@ -0,0 +1,207 @@
+
+/obj/item/weapon/reagent_containers/hypospray/autoinjector/pouch_auto/update_icon()
+	if(reagents.total_volume > 0)
+		icon_state = "[icon_colour]1"
```
As above, why doesn't `initial(icon_state)` do?

PurpleMartinJCK replied:
Here it doesn't work because the colour of the autoinjector is changed shortly after it's created. The initial icon state isn't the right colour.

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/storage.dm:
```diff
@@ -341,6 +342,8 @@
 	prepare_ui()
 
 	if(startswith)
+		if(start_rand)
```
This doesn't really matter one way or another, but I fail to see the purpose.

PurpleMartinJCK commented:
@PsiOmegaDelta fixed
